web/yo/app/scripts/services/stringutils.js

```diff
@@ -879,206 +879,174 @@ angular.module('oncokbApp')
             gene.mutations = [];
             gene.curators = [];
             gene.transcripts = [];
             geneData.curators.asArray().forEach(function(e) {
                 var _curator = {};
-                _curator = combineData(_curator, e, ['name', 'email'], excludeObsolete, excludeComments, onlyReviewedContent);
+                _curator = combineData(_curator, e, ['name', 'email'], excludeComments, onlyReviewedContent);
                 gene.curators.push(_curator);
             });
             geneData.transcripts.asArray().forEach(function(e) {
                 var _transcript = {};
-                _transcript = combineData(_transcript, e, ['isoform_override', 'gene_name', 'dmp_refseq_id', 'ccds_id'], excludeObsolete, excludeComments, onlyReviewedContent);
+                _transcript = combineData(_transcript, e, ['isoform_override', 'gene_name', 'dmp_refseq_id', 'ccds_id'], excludeComments, onlyReviewedContent);
                 gene.transcripts.push(_transcript);
             });
             geneData.mutations.asArray().forEach(function(e) {
-                if (onlyReviewedContent && e.name_review.get('added') == true) return;
-                if (!(excludeObsolete && e.name_eStatus && e.name_eStatus.has('obsolete') && e.name_eStatus.get('obsolete') === 'true') && (!excludeRedHands || e.oncogenic_eStatus.get('curated') !== false)) {
-                    var _mutation = {};
-                    _mutation.tumors = [];
-                    _mutation.effect = {};
-                    _mutation = combineData(_mutation, e, ['name'], excludeObsolete, excludeComments, onlyReviewedContent);
-                    // This is a weird way to do, but due to time constraint, this has to be implemented in this way.
-                    // I assigned shortSummary estatus for oncogenic and oncogenic estatus to mutation effect,
-                    // so there is no need to check excludeObsolete since I did outside of combinedata.
-                    if (!(excludeObsolete && e.shortSummary_eStatus && e.shortSummary_eStatus.has('obsolete') && e.shortSummary_eStatus.get('obsolete') === 'true')) {
-                        _mutation = combineData(_mutation, e, ['shortSummary', 'oncogenic'], false, excludeComments, onlyReviewedContent);
-                    }
-                    if (!(excludeObsolete && e.oncogenic_eStatus && e.oncogenic_eStatus.has('obsolete') && e.oncogenic_eStatus.get('obsolete') === 'true')) {
-                        _mutation = combineData(_mutation, e, ['description', 'short', 'effect'], excludeObsolete, excludeComments, onlyReviewedContent);
-                        if (e.effect_uuid) {
-                            _mutation.effect_uuid = validUUID(e.effect_uuid);
-                        }
-                        _mutation.effect_review = getReview(e.effect_review);
+                if (onlyReviewedContent && e.name_review.get('added') == true || !onlyReviewedContent && e.name_review.get('removed') == true) return;
```

Comment:
`&& e.name_review.get('removed') == true` why we have new logic?

Reply:
This logic is to apply onlyReviewedContent option to removed sections, which were somehow not included before. For instance, if a tumor is removed but not approved yet, we should still import this tumor if onlyReviewedContent is set to true. In the other way, that tumor will not be imported if onlyReviewedContent is set to false.